lib/entities/game.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+class Game
+  attr_reader :difficulty, :total_attempts, :used_attempts, :total_hints, :used_hints, :secret_code
+
+  SECRET_CODE_LENGTH = 4
+  ELEMENT_VALUE_RANGE = (1..6).freeze
+
+  DELIMITER = ''
+  EXACT_MATCH = '+'
+  NUMBER_MATCH = '-'
+
+  def initialize(difficulty)
+    @difficulty = difficulty[:level]
+    @total_attempts = difficulty[:attempts]
+    @used_attempts = 0
+    @total_hints = difficulty[:hints]
+    @used_hints = 0
+    @secret_code = generate
+    @shuffled_code = @secret_code.shuffle
+  end
+
+  def use_hint
+    increment_used_hints
+    @shuffled_code.shift
+  end
+
+  def hints_available?
+    @used_hints >= @total_hints
+  end
+
+  def win?(guess_code)
```

Comment:
too strange that you compare array of numbers? maybe it's better to compare number with number or even string with string